[4.0] Redo of custom elements tabs

dgrammatiko opened the pull request:
Pull Request for many Issues originating from the tabs script

fixes:
https://github.com/joomla/joomla-cms/issues/33229
https://github.com/joomla/joomla-cms/issues/34492
https://github.com/joomla/joomla-cms/issues/31583

### This should be a Release Blocker

### Summary of Changes
- The logic was rewritten from scratch
- Same funtionality
- Better code
- Support for dynamically added tabs
- A11y fixes (based on the WAI-ARIA recommendations)

Changes of the source js/css can be found: https://github.com/joomla-projects/custom-elements/pull/180

### Testing Instructions
Test each and every form in the backend that has tabs


### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request

Tabs View:
<img width="1548" alt="Screenshot 2021-07-17 at 14 06 02" src="https://user-images.githubusercontent.com/3889375/126036338-88602c4a-ad5f-4ea3-a4c0-173b84ac9105.png">

<img width="1552" alt="Screenshot 2021-07-17 at 14 06 14" src="https://user-images.githubusercontent.com/3889375/126036343-5d28b741-9ac1-4aa1-9e76-f9d8d7ba8980.png">

Shrink to accordion:
<img width="596" alt="Screenshot 2021-07-17 at 14 07 03" src="https://user-images.githubusercontent.com/3889375/126036347-f622a888-f77c-47d3-9475-5c913e7270ad.png">

<img width="596" alt="Screenshot 2021-07-17 at 14 07 03" src="https://user-images.githubusercontent.com/3889375/126036351-94ad4239-f659-43c4-9597-2fc1157fced2.png">


### Documentation Changes Required

- Added the recall and breakpoint attributes to the expected values for the creation of the tabs (UiTabs::)
- The `section` element (the tab content) is renamed to `joomla-tab-element`

brianteeman commented on libraries/src/HTML/Helpers/UiTab.php:
```diff
@@ -79,7 +82,7 @@ public static function endTabSet()
 	 * @param   string  $selector  Identifier of the panel.
 	 * @param   string  $id        The ID of the div element
-	 * @param   string  $title     The title text for the new UL tab
+	 * @param   string  $title     The title text for the button
```
button ??

dgrammatiko replied:
It's the tab/accordion button that switches the tab/accordion (the navigation part)

brianteeman commented on plugins/content/pagebreak/pagebreak.php:
```diff
@@ -221,7 +221,7 @@ public function onContentPrepare($context, &$row, &$params, $page = 0)
 				if ($style === 'tabs')
 				{
-					$t[] = (string) HTMLHelper::_('uitab.startTabSet', 'myTab', array('active' => 'article' . $row->id . '-' . $style . '0'));
+					$t[] = (string) HTMLHelper::_('uitab.startTabSet', 'myTab', array('active' => 'article' . $row->id . '-' . $style .'0', 'recall' => true, 'breakpoint' => 768));
```
```suggestion
					$t[] = (string) HTMLHelper::_('uitab.startTabSet', 'myTab', array('active' => 'article' . $row->id . '-' . $style . '0', 'recall' => true, 'breakpoint' => 768));
```

dgrammatiko committed:
Update plugins/content/pagebreak/pagebreak.php

Co-authored-by: Brian Teeman <brian@teeman.net>

richard67 commented on libraries/src/HTML/Helpers/UiTab.php:
```diff
@@ -54,12 +54,15 @@ public static function startTabSet($selector = 'myTab', $params = array())
+		$recall = isset($params['recall']) ? 'recall' : '';
+		$breakpoint  = isset($params['breakpoint']) ? 'breakpoint="' . $params['breakpoint'] . '"' : '';
+
+		if (!isset($params['breakpoint']) && $breakpoint === '') {
```
```suggestion
		if (!isset($params['breakpoint']) && $breakpoint === '')
		{
```
Fix PHP CS error in Drone.

richard67 commented on libraries/src/HTML/Helpers/UiTab.php:
```diff
@@ -54,12 +54,15 @@ public static function startTabSet($selector = 'myTab', $params = array())
+		}
 
-		return '<joomla-tab id="' . $selector . '" orientation="' . $orientation . '" recall>';
+		return '<joomla-tab id="' . $selector . '" orientation="' . $orientation . '" ' . $recall .' ' . $breakpoint . '>';
```
```suggestion
		return '<joomla-tab id="' . $selector . '" orientation="' . $orientation . '" ' . $recall . ' ' . $breakpoint . '>';
```
Fix PHP CS error in Drone.

dgrammatiko committed:
Update libraries/src/HTML/Helpers/UiTab.php

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

dgrammatiko committed:
Update libraries/src/HTML/Helpers/UiTab.php

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

richard67 commented:
@dgrammatiko Now PHPCS is fine, but npm is failing in Drone. Could that be the same problem as with your other PR?

https://ci.joomla.org/joomla/joomla-cms/45867/1/1

```
npm ci --unsafe-perm
npm ERR! code ENOENT
npm ERR! syscall spawn git
npm ERR! path git
npm ERR! errno -2
npm ERR! enoent Error while executing:
npm ERR! enoent undefined ls-remote -h -t https://github.com/joomla-projects/custom-elements.git
npm ERR! enoent 
npm ERR! enoent 
npm ERR! enoent spawn git ENOENT
npm ERR! enoent This is related to npm not being able to find a file.
```

dgrammatiko committed:
Fix the permission tab styling

dgrammatiko commented:
> Could that be the same problem as with your other PR?

Yes, it's exactly the same issue: the node alpine container doesn't have the git installed.
I guess this limits the testers to only the ones that can git pull, etc (pretty sure also the created packages are broken)

richard67 commented:
> I guess this limits the testers to only the ones that can git pull, etc (pretty sure also the created packages are broken)

@dgrammatiko Correct.

Quy commented on administrator/components/com_banners/tmpl/banner/edit.php:
```diff
@@ -29,7 +29,7 @@
 
 	<div class="main-card">
-		<?php echo HTMLHelper::_('uitab.startTabSet', 'myTab', array('active' => 'details')); ?>
+		<?php echo HTMLHelper::_('uitab.startTabSet', 'myTab', array('active' =>'details', 'recall' => true, 'breakpoint' => 768)); ?>
```
```suggestion
		<?php echo HTMLHelper::_('uitab.startTabSet', 'myTab', array('active' => 'details', 'recall' => true, 'breakpoint' => 768)); ?>
```

Quy replied:
Please fix the same with other instances.

dgrammatiko committed:
Update administrator/components/com_banners/tmpl/banner/edit.php

Co-authored-by: Quy <quy@fluxbb.org>

Quy commented:
This displays quickly then disappears.

![joomla-stats](https://user-images.githubusercontent.com/368084/126043453-d8347a84-395a-4ff5-8951-005bdf34caff.jpg)

Quy commented:
Text should be left-aligned.

![permissions](https://user-images.githubusercontent.com/368084/126043527-a8a6c117-8453-4947-9b84-436f1e5f8e87.jpg)

dgrammatiko commented:
> This displays quickly then disappears.

did you test the https://github.com/joomla/joomla-cms/pull/34781 before? Sounds like this is comming from there

Quy commented:
This is a clean install from the download page of this PR.

dgrammatiko commented:
> This is a clean install from the download page of this PR.

Ahh, forgot that George already merged a PR for alerts that wasn't quite ready, treat anything alerts related here as irrelevant

Quy commented:
Permissions is selected.

![global-configuration-permissions](https://user-images.githubusercontent.com/368084/126043608-0f3611cb-26af-4b83-beb1-d19461179a60.jpg)

Go to a component. No tab is selected by default.

![global-configuration](https://user-images.githubusercontent.com/368084/126043575-27959a30-e764-4975-93c5-3acf313f99bc.jpg)

dgrammatiko committed:
fixes

- media manager edit page
- - module
- - loading issues (removes a todo)
- - Crop is not initialising correctly
- - apply doesn’t work as expected
- style vertical tabs so text is always at the begining

dgrammatiko commented:
@Quy should ok now
<img width="1386" alt="Screenshot 2021-07-17 at 20 09 02" src="https://user-images.githubusercontent.com/3889375/126046130-078fbf58-0100-43c8-aba9-b35c784b6864.png">

Quy commented:
Go to Global Configuration.
Click on one of the components.
Styling issue on the first tab.

![tab-style](https://user-images.githubusercontent.com/368084/126046673-58a2e4b3-6028-4607-b261-1001e3b0f050.jpg)

Quy commented:
In mobile view, text should be left-aligned. The vertical ellipse is not visible against the blue background.

![tab-mobile](https://user-images.githubusercontent.com/368084/126046775-5c1f7309-5231-4c0c-84e2-1f7c00fe5ba2.jpg)

Quy commented:
Go to System > Administrator Templates > Atum Details and Files

![tab-editor](https://user-images.githubusercontent.com/368084/126046924-bb8f6c0a-996b-4248-9535-faa3a27213e6.jpg)

![tab-create-overrides](https://user-images.githubusercontent.com/368084/126046929-38960bcc-f228-4b57-8bf8-7eda561bd6d6.jpg)

dgrammatiko committed:
styling

dgrammatiko committed:
fetch the newer code

Quy commented:
Edit an article.
Hover over `Images and Links` tab.
`Images and Links` becomes bold causing words/tabs to the right of this tab to shift.

Quy commented:
Will PR #31419 be obsolete because of this PR?

dgrammatiko committed:
Fix the focused elements

dgrammatiko commented:
> The label becomes bold causing words/tabs to the right of this tab to shift.

Fixed

dgrammatiko commented:
> Will PR #31419 be obsolete because of this PR?

With the provided code (modified to use more attributes):
```html
<?= HTMLHelper::_('uitab.startTabset', 'tabs1', ['active' => 'tabA', 'view' => 'tabs', 'orientation' => 'horizontal']) ?>
<?= HTMLHelper::_('uitab.addTab', 'tabs1', 'tabs11', 'tabs11') ?>
Horizontal tab 1

<?= HTMLHelper::_('uitab.startTabset', 'tabs2', ['active' => 'tabs21', 'view' => 'tabs', 'orientation' => 'vertical']) ?>
<?= HTMLHelper::_('uitab.addTab', 'tabs2', 'tabs21', 'tabs21') ?>
Horizontal &gt; Vertical tab 1

<?= HTMLHelper::_('uitab.startTabset', 'tabs3', ['active' => 'tabs31', 'view' => 'tabs', 'orientation' => 'horizontal']) ?>
<?= HTMLHelper::_('uitab.addTab', 'tabs3', 'tabs31', 'tabs31') ?>
Horizontal &gt; Vertical &gt; Horizontal tab 1
<?= HTMLHelper::_('uitab.endTab') ?>
<?= HTMLHelper::_('uitab.addTab', 'tabs3', 'tabs32', 'tabs32') ?>
Horizontal &gt; Vertical &gt; Horizontal tab 2
<?= HTMLHelper::_('uitab.endTab') ?>
<?= HTMLHelper::_('uitab.endTabset') ?>

<?= HTMLHelper::_('uitab.endTab') ?>

<?= HTMLHelper::_('uitab.addTab', 'tabs2', 'tabs22', 'tabs22') ?>
Horizontal &gt; Vertical tab 2
<?= HTMLHelper::_('uitab.endTab') ?>

<?= HTMLHelper::_('uitab.endTabset') ?>

<?= HTMLHelper::_('uitab.endTab') ?>

<?= HTMLHelper::_('uitab.addTab', 'tabs1', 'tabs12', 'tabs12') ?>
Horizontal tab 2
<?= HTMLHelper::_('uitab.endTab') ?>

<?= HTMLHelper::_('uitab.endTabset') ?>

```

I get: 
<img width="1763" alt="Screenshot 2021-07-17 at 23 12 14" src="https://user-images.githubusercontent.com/3889375/126049484-9a23caf4-6d21-422b-834e-09b10347fb81.png">

(which is the expected output)

brianteeman commented:
testing the changing of permissions for an article. After each change there is the following javascript error

```
joomla-tab.min.js?0.0.44:1 
Uncaught TypeError: t.getAttribute is not a function
    at HTMLElement.removeNavs (joomla-tab.min.js?0.0.44:1)
    at joomla-tab.min.js?0.0.44:1
    at Array.map (<anonymous>)
    at HTMLElement.onMutation (joomla-tab.min.js?0.0.44:1)
```

dgrammatiko committed:
CS

brianteeman commented:
I have tested this item :red_circle: unsuccessfully on 4f0407cf129312ed884b1dd8be1568bb6c146b14<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34813">issues.joomla.org/tracker/joomla-cms/34813</a>.</sub>

dgrammatiko committed:
fixes

- console errors
- wrong state activation

dgrammatiko commented:
@brianteeman was it only these console errors or there are more bugs?

btw I already pushed some fixes

brianteeman commented:
Yes - I stopped at that point to do other stuff. retesting now

brianteeman commented:
js issue is fixed.

there is however a css issue

before this PR
![image](https://user-images.githubusercontent.com/1296369/126075291-171deef6-a099-4057-9cad-80a26ca619a6.png)


with this PR
![image](https://user-images.githubusercontent.com/1296369/126075293-21df965f-c6e5-4d01-aa4f-98b9854af5cf.png)

dgrammatiko committed:
Restore vertical dots

dgrammatiko commented:
> there is however a css issue

@brianteeman should be ok now

dgrammatiko committed:
Keep the same css var

brianteeman commented:
confirm that the tree is fixed now.

Spotted something different not which I think is something to do with focus-visible
![tabs](https://user-images.githubusercontent.com/1296369/126076336-e46e5f9c-3452-4ec9-9e03-aedb24175be7.gif)

dgrammatiko committed:
Skip focus on init

dgrammatiko commented:
> Spotted something different not which I think is something to do with focus-visible

The focus should work as expected now

dgrammatiko committed:
more styling fixes

brianteeman commented:
> The focus should work as expected now

Still needs fixing here 
![image](https://user-images.githubusercontent.com/1296369/126078877-51ea57c5-7227-42b7-8d2a-4b076450dd05.png)

Quy commented:
No background color on the active tab. Padding is reduced.

BEFORE PR:
![34813-tinymce-before](https://user-images.githubusercontent.com/368084/126079102-c12e467b-3b4b-4d78-b628-c80b5ac61aec.jpg)

AFTER PR:
![34813-tinymce-after](https://user-images.githubusercontent.com/368084/126079105-9380c5ce-a43d-46c6-8128-dd44f805d808.jpg)

dgrammatiko committed:
Fix focus

- Image editing fixes
- Rework crop bug

dgrammatiko commented:
> Still needs fixing here

That was another logic, should be ok now, also fixes another bug with image editing

dgrammatiko committed:
Fix plugins edit margin

dgrammatiko committed:
active nav background

dgrammatiko commented:
@Quy done

dgrammatiko committed:
Well, padding is p, margin is m...

Quy commented:
![34821-permissions](https://user-images.githubusercontent.com/368084/126080205-73dcf878-5ac5-4033-b857-30dda039ed92.jpg)

Quy commented:
When editing/closing an image.

![34821-undefined](https://user-images.githubusercontent.com/368084/126080229-8ea6a29c-109d-4e8c-8703-bba35c268ca9.jpg)

![34821-media-js](https://user-images.githubusercontent.com/368084/126080233-ee4ae891-9a41-45f4-a1f8-fa763ae8a9c1.jpg)

Quy commented:
Edit an article. See padding less with PR.

BEFORE:
![34821-content-pre](https://user-images.githubusercontent.com/368084/126080352-27b371df-1a58-4097-a0da-a1f01bbc71a7.jpg)
![34821-images-pre](https://user-images.githubusercontent.com/368084/126080351-6fc0cd00-600f-4d04-a5d3-601e1ba81c6c.jpg)

AFTER:
![34821-content-after](https://user-images.githubusercontent.com/368084/126080364-06d124e0-d61c-48f7-a1fe-989e31f78184.jpg)
![34821-images-after](https://user-images.githubusercontent.com/368084/126080363-ccbca6be-6020-4893-b6a6-7327b619bbda.jpg)

dgrammatiko committed:
Fix crop

dgrammatiko commented:
> When editing/closing an image.

Fixed.

Reviewing all the padding/margin values

dgrammatiko committed:
Fix styling inconsistencies

Quy commented:
Go to Media.
Edit an image.
Click Resize tab.
It is random when the image won't display.
Click Close button.
Edit the same image.
Image displays find under all tabs.

![34813-media](https://user-images.githubusercontent.com/368084/126240358-92558f38-1dc3-418c-b189-9e1dd59fecad.jpg)

dgrammatiko committed:
Changes

dgrammatiko committed:
Reverting the media/image editor changes here

Meaning that in this PR the Image editing is broken!!!

dgrammatiko commented:
A note here for testers: 

## The image editing is broken here but there's a new PR for that: https://github.com/joomla/joomla-cms/pull/34885

dgrammatiko committed:
Merge gone wrong

Quy commented:
Click User Menu > Accessibility Settings
The `Accessibility Setting`s tab is not selected.

Quy commented on administrator/components/com_installer/tmpl/install/default.php:
```diff
@@ -51,7 +51,7 @@
 
 					<?php if ($tabs) : ?>
-						<?php echo HTMLHelper::_('uitab.startTabSet', 'myTab', ['active' => $tabs[0]['name'] ?? '']); ?>
+						<?php echo HTMLHelper::_('uitab.startTabSet', 'myTab', ['active' => $tabs[0]['name'] ??'', 'recall' => true, 'breakpoint' => 768]); ?>
```
```suggestion
						<?php echo HTMLHelper::_('uitab.startTabSet', 'myTab', ['active' => $tabs[0]['name'] ?? '', 'recall' => true, 'breakpoint' => 768]); ?>
```

dgrammatiko committed:
Update administrator/components/com_installer/tmpl/install/default.php

Co-authored-by: Quy <quy@fluxbb.org>

dgrammatiko committed:
Activate tab from hash

dgrammatiko commented:
> The Accessibility Settings tab is not selected.

Good catch,  fixed with https://github.com/joomla/joomla-cms/pull/34813/commits/bae7dfe2b807cbcf4f7175da5b60eea71547568c

dgrammatiko committed:
Fetch latest build (CS changes)

Quy commented:
Click User Menu > Accessibility Settings
Accessibility Settings tab is selected.
Click Close button.
Click User Menu > Accessibility Settings
Tab outlined. Click anywhere on the screen to refresh/fix the outline.

Firefox:
![34813-firefox](https://user-images.githubusercontent.com/368084/126842482-f376fbf3-e924-441e-b0ba-351f9f1e284e.jpg)

Chrome:
![34813-chrome](https://user-images.githubusercontent.com/368084/126842485-2500b8b5-4993-4e77-b187-d44484cf551c.jpg)

brianteeman commented:
didnt i report that before?

dgrammatiko committed:
Don't focus the hash activated tab

dgrammatiko commented:
> didnt i report that before?

You did and I fixed it, but I added the hash activation part that I forgot and I forgot to pass the `false` param.
Anyways it's fixed with https://github.com/joomla/joomla-cms/pull/34813/commits/cbce7b1113f4802543589a1eef8a29134bb7c281

Quy commented on administrator/components/com_plugins/tmpl/plugin/edit.php:
```diff
@@ -32,11 +32,11 @@
 		<?php echo HTMLHelper::_('uitab.addTab', 'myTab', 'general', Text::_('COM_PLUGINS_PLUGIN')); ?>
 
-		<div class="row mt-2">
+		<div class="row mt-2 ps-4">
```
```suggestion
		<div class="row mt-2">
```

![34813-plugin](https://user-images.githubusercontent.com/368084/126847435-b4eca67e-2b14-42d8-a0de-57a51256fb07.jpg)

dgrammatiko committed:
Update administrator/components/com_plugins/tmpl/plugin/edit.php

Co-authored-by: Quy <quy@fluxbb.org>

Quy commented:
I have tested this item :white_check_mark: successfully on 8f79a32a776e250e34408a9d12cd276102bca483<br /><br />THANK YOU!!!!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34813">issues.joomla.org/tracker/joomla-cms/34813</a>.</sub>

dgrammatiko commented:
@Quy thank you for all the tests here  🙌

ceford commented:
I have tested this item :white_check_mark: successfully on 8f79a32a776e250e34408a9d12cd276102bca483<br /><br />Went through most of the edit form and did not find any problems.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34813">issues.joomla.org/tracker/joomla-cms/34813</a>.</sub>

richard67 commented:
Maintainers please note that this PR should not be merged as it is. It needs to merge the corresponding upstream PR and make a new tag or release upstream and then change here to let npm use that. It will not need new tests after that if the new upstream tag or release point to the same upstream commit as used by this PR here now.

dgrammatiko commented:
There's one more step, so, here's the whole procedure:
- on the Custom Elements Repo the https://github.com/joomla-projects/custom-elements/pull/180 needs to be merged
- A new patch version (0.2.0) needs to be released
- This PR needs to update the package.json and package.lock
- The PR #34885 needs to merge immediately after (assuming that what I did there is acceptable)

But since this PR is using the commit hash no further tests will be needed if the PR https://github.com/joomla-projects/custom-elements/pull/180 is merged without any changes, that was the reason I used the commit hashes here in the first place. Anyways these are @wilsonge's problems  😀

ceford commented:
Something I neglected to check: the Article edit form has 12 html validation errors.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34813">issues.joomla.org/tracker/joomla-cms/34813</a>.</sub>

dgrammatiko commented:
> Something I neglected to check: the Article edit form has 12 html validation errors.

Can you post the report here?

dgrammatiko committed:
Reverting the crop options changes 

This shouldn't affect the test results here but it's easier to undo the changes here than on the other PR

ceford commented:
> > Something I neglected to check: the Article edit form has 12 html validation errors.
> 
> Can you post the report here?

It is probably best to paste the page source into the w3C validator:

https://validator.w3.org/#validate_by_input

30 Errors on the article edit page. It is not easy to copy them into a message.

dgrammatiko commented:
@ceford you're validating as HTML5, right?
<img width="768" alt="Screenshot 2021-07-24 at 14 28 10" src="https://user-images.githubusercontent.com/3889375/126868430-453ac0f5-031a-451f-ac92-191919c1974f.png">

ceford commented:
Yes - I use the Firefox Web Developer plugin. It is just two clicks to invoke the w3c validator with default settings. I get 30 Errors for the Article edit page with no patches applied. It returns a blank page if I have Joomla debug enabled.

dgrammatiko commented:
> I get 30 Errors for the Article edit page with no patches applied

Yes, I could see also the errors but none of them is about tabs

>  It returns a blank page if I have Joomla debug enabled.

Probably because the page is extremely heavy with debug on (too many elements)

wilsonge commented:
@dgrammatiko i've released the custom elements repo. If you could use the tagged version here then one last test by someone just to do final validation would be excellent.

dgrammatiko committed:
Use the release

dgrammatiko committed:
SCSS CS

dgrammatiko committed:
Tests

Quy commented on administrator/components/com_banners/tmpl/banner/edit.php:
```diff
@@ -29,7 +29,7 @@
 
 	<div class="main-card">
-		<?php echo HTMLHelper::_('uitab.startTabSet', 'myTab', array('active' => 'details')); ?>
+		<?php echo HTMLHelper::_('uitab.startTabSet', 'myTab', array('active' => 'details', 'recall' => true, 'breakpoint' => 768)); ?>
```
```suggestion
		<?php echo HTMLHelper::_('uitab.startTabSet', 'myTab', ['active' => 'details', 'recall' => true, 'breakpoint' => 768]); ?>
```

dgrammatiko committed:
Update administrator/components/com_banners/tmpl/banner/edit.php

Co-authored-by: Quy <quy@fluxbb.org>

dgrammatiko committed:
tests

dgrammatiko committed:
c

dgrammatiko committed:
🤬

Quy commented on plugins/content/pagebreak/pagebreak.php:
```diff
@@ -221,7 +221,7 @@ public function onContentPrepare($context, &$row, &$params, $page = 0)
 				if ($style === 'tabs')
 				{
-					$t[] = (string) HTMLHelper::_('uitab.startTabSet', 'myTab', array('active' => 'article' . $row->id . '-' . $style . '0'));
+					$t[] = (string) HTMLHelper::_('uitab.startTabSet', 'myTab', array('active' => 'article' . $row->id . '-' . $style . '0', 'view' => 'tabs'));
```
```suggestion
					$t[] = (string) HTMLHelper::_('uitab.startTabSet', 'myTab', ['active' => 'article' . $row->id . '-' . $style . '0', 'view' => 'tabs']);
```

dgrammatiko committed:
Update plugins/content/pagebreak/pagebreak.php

Co-authored-by: Quy <quy@fluxbb.org>

dgrammatiko committed:
nope

dgrammatiko committed:
tests

dgrammatiko commented:
Tests should be ok with https://github.com/joomla-projects/joomla-browser/pull/219

wilsonge commented:
OK Can we have one final test of the code here now we're pulling in the stable custom elements repo and I'll get everything merged through.

richard67 commented:
> 
> 
> OK Can we have one final test of the code here now we're pulling in the stable custom elements repo and I'll get everything merged through.

@wilsonge And what about failing system tests?

wilsonge commented:
The PR above needs to be merged according to @dgrammatiko - that would make the 4.0-dev fail if i merged that first - so i'll merge both at the same time and should be fine. I can debug further if anything else still fails.

richard67 commented:
It's late here already and I'm tired. Maybe @Quy can test again?

Quy commented:
Border thickness is thinner.

Before PR:
![34813-before](https://user-images.githubusercontent.com/368084/126914342-a6148c10-fd0e-42a9-a5cd-74865ef42a42.jpg)

After PR (thickness difference between first row and second row):
![34813-after](https://user-images.githubusercontent.com/368084/126914346-bb77a79a-8587-4f9b-a161-fec8c268cefb.jpg)

![34813-after-2nd-row](https://user-images.githubusercontent.com/368084/126914429-088f277f-0571-4c7d-a136-8745130602b8.jpg)

dgrammatiko committed:
active border 1px...

dgrammatiko commented:
@Quy pixel restored

Quy commented:
There is a large gap at the bottom of the tab.

Before PR:
![34813-global-before](https://user-images.githubusercontent.com/368084/126932496-dc65148d-86ef-40a0-8a0b-cfebad57b082.jpg)

After PR:
![34813-global-after](https://user-images.githubusercontent.com/368084/126932503-402d1c22-e694-49b5-9eb0-d0af82bec80c.jpg)

Quy commented:
Fieldset left/right borders missing.

Before PR:
![34813-messages-before](https://user-images.githubusercontent.com/368084/126932776-5686bc66-67e8-41d2-ab92-26fdc3980936.jpg)

After PR:
![34813-messages-after](https://user-images.githubusercontent.com/368084/126932779-d79c0d4c-9c31-43d7-b0fd-a9c116b31b38.jpg)

dgrammatiko committed:
style fixes

dgrammatiko commented:
@Quy should be ok now

Quy commented:
I have tested this item :white_check_mark: successfully on c75abd12b564df4114d89620da7ddc64ca7cddb8<br /><br />THANK YOU AGAIN!!!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34813">issues.joomla.org/tracker/joomla-cms/34813</a>.</sub>

Quy commented:
I don't see errors in the Console.

System > Install > Extensions > Install from Web

![34813-install](https://user-images.githubusercontent.com/368084/127017344-48cc3dde-6588-48e1-b5e6-81d5a3cacc26.jpg)

richard67 commented:
SCSS style linter complains: https://ci.joomla.org/joomla/joomla-cms/46132/1/20

dgrammatiko committed:
SCSS cs && Web Installer patch

dgrammatiko commented:
> System > Install > Extensions > Install from Web

Web Installer fixed

> SCSS style linter complains: https://ci.joomla.org/joomla/joomla-cms/46132/1/20

CS fixed

Quy commented:
I have tested this item :white_check_mark: successfully on c75abd12b564df4114d89620da7ddc64ca7cddb8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34813">issues.joomla.org/tracker/joomla-cms/34813</a>.</sub>

wilsonge commented:
Thanks!

dgrammatiko commented:
Thanks!

dgrammatiko commented:
@wilsonge you need to merge the https://github.com/joomla-projects/joomla-browser/pull/219 and also the https://github.com/joomla/joomla-cms/pull/34885

The  https://github.com/joomla-projects/joomla-browser/pull/219 will fix the failing tests
